src/main/scala/chisel3/util/SRAM.scala

```diff
@@ -79,14 +79,18 @@ class MemoryReadWritePort[T <: Data](tpe: T, addrWidth: Int, masked: Boolean) ex
   * @param numReadPorts The number of read ports
   * @param numWritePorts The number of write ports
   * @param numReadwritePorts The number of read/write ports
+  * @param masked Whether the memory is write masked
+  * @param underlying Points to SRAM that instantiates this interface; None by
+  *                   default if this was instantiated directly
   */
 class SRAMInterface[T <: Data](
   memSize:           BigInt,
   tpe:               T,
   numReadPorts:      Int,
   numWritePorts:     Int,
   numReadwritePorts: Int,
-  masked:            Boolean = false)
+  masked:            Boolean = false,
+  val underlying:    Option[HasTarget] = None)
```

Comment:
@jackkoenig i am concerned that the `SRAMInterface.underlying` can be set to any random `HasTarget`… is that ok? Should i make a private setter or something? Or is that overkill

Reply:
It's a fair point and making it a private setter would be more conservative. It would then be possible to make the underlying not propagate when the type is cloned which might be useful. There's still possibly some utility in having it be public and settable by users since someone could have their own implementation of SRAM that returns SRAMInterface and have the underlying point to their own `HasTarget` (maybe a module?).

In any case, I think going with the more conservative option is probably best since we can always open up the API later, so yeah go ahead and make it a private setter, something like:
```scala
  private[chisel3] var _underlying: Option[HasTarget] = None
  /** Useful ScalaDoc */
  def underlying: Option[HasTarget] = _underlying
```